src/dir.cr

```diff
@@ -65,7 +60,7 @@ class Dir
   end
 
   def each_entry
-    EntryIterator.new(self)
+    @dir
```

Comment:
`each_entry` should return a new iterator each it is called, no?

Reply:
If you look at the behaviour of the old iterator, this wasn't how it worked. The whole `Dir` class is a very poor iterator, the `each_entry` iterator simply wrapped it up into a real `Iterator`, but the state was still in `Dir`. This, while confusing, keeps the old `Dir` semantics.

Reply:
I plan to clean up `Dir` after merging this PR, and make it a module with entirely class methods.

Reply:
I disagree with the old semantics, but I do agree that this PR should not change the semantics. So let it be.

Reply:
I think everyone sane disagrees with the old semantics!